app/src/main/java/com/fastlink/zekrapp/viewModel/ViewModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.fastlink.zekrapp.viewModel
+
+import android.util.Log
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import com.fastlink.zekrapp.appData.data.categoriesData
+import com.fastlink.zekrapp.appData.model.CategoryModel
+import com.fastlink.zekrapp.appData.model.ZekrModel
+import com.fastlink.zekrapp.appData.data.getCateories
+import com.fastlink.zekrapp.appData.data.getZekrs
+import com.fastlink.zekrapp.appData.data.zekrs
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+    init {
+        getZekrs()
+        getCateories()
+    }
+
+    private var _categories = mutableStateOf(getAllCategories())
+    val zekirNumber = mutableIntStateOf(0)
+    val zekirCounter = mutableIntStateOf(0)
+    var zekirs :  List<ZekrModel> = emptyList()
+
+
+
+    fun onClicked(scope : CoroutineScope) {
+        if (zekirCounter.intValue < zekirs[zekirNumber.intValue].counterNumber) {
+            zekirCounter.intValue++
+            if (zekirCounter.intValue == zekirs[zekirNumber.intValue].counterNumber) {
+                scope.launch {
+                    delay(1500)
+                    if (zekirNumber.intValue < zekirs.size - 1) {
+                        zekirCounter.intValue = 0
+                        zekirNumber.intValue++
+//                        scope.launch {
```

Comment:
Ahmed already mentioned dead code, here and in a lot of other places

Reply:
sorry for dead code i will remove it, kak ahmad mentioned to use `SnackBar` instead of `Toast`